File: db4oj/core/src/com/db4o/internal/ObjectContainerFactory.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */

package com.db4o.internal;

import com.db4o.*;
import com.db4o.config.*;
import com.db4o.ext.*;
import com.db4o.internal.config.*;


public class ObjectContainerFactory {
	
	public static EmbeddedObjectContainer openObjectContainer(EmbeddedConfiguration config, String databaseFileName) throws OldFormatException {		
		Configuration legacyConfig = Db4oLegacyConfigurationBridge.asLegacy(config);		
		Config4Impl.assertIsNotTainted(legacyConfig);
		
		emitDebugInfo();		
		EmbeddedObjectContainer oc = new IoAdaptedObjectContainer(legacyConfig, databaseFileName);	
		((EmbeddedConfigurationImpl)config).applyConfigurationItems(oc);
		Messages.logMsg(legacyConfig, 5, databaseFileName);
		return oc;
	}

	private static void emitDebugInfo() {
	    if (Deploy.debug) {
			System.out.println("db4o Debug is ON");
			if (!Deploy.flush) {
				System.out.println("Debug option set NOT to flush file.");
			}
		}
    }
}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/ObjectContainerFactory.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Config;

namespace Db4objects.Db4o.Internal
{
	public class ObjectContainerFactory
	{
		/// <exception cref="Db4objects.Db4o.Ext.OldFormatException"></exception>
		public static IEmbeddedObjectContainer OpenObjectContainer(IEmbeddedConfiguration
			 config, string databaseFileName)
		{
			IConfiguration legacyConfig = Db4oLegacyConfigurationBridge.AsLegacy(config);
			Config4Impl.AssertIsNotTainted(legacyConfig);
			EmitDebugInfo();
			IEmbeddedObjectContainer oc = new IoAdaptedObjectContainer(legacyConfig, databaseFileName
				);
			((EmbeddedConfigurationImpl)config).ApplyConfigurationItems(oc);
			Db4objects.Db4o.Internal.Messages.LogMsg(legacyConfig, 5, databaseFileName);
			return oc;
		}

		private static void EmitDebugInfo()
		{
		}
	}
}
